redis/src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -8,18 +8,30 @@ trait SortedSets {
   final val bzPopMax = RedisCommand("BZPOPMAX", Tuple2(DurationSecondsInput, NonEmptyList(StringInput)), ChunkOutput)
   final val bzPopMin = RedisCommand("BZPOPMIN", Tuple2(DurationSecondsInput, NonEmptyList(StringInput)), ChunkOutput)
 
-  // TODO: represent two possible outputs as per the documentation
-  final val zAdd = RedisCommand(
-    "ZADD",
-    Tuple5(
-      StringInput,
-      OptionalInput(UpdateInput),
-      OptionalInput(ChangedInput),
-      OptionalInput(IncrementInput),
-      NonEmptyList(MemberScoreInput)
-    ),
-    LongOutput
-  )
+  final def zAdd =
+    RedisCommand(
+      "ZADD",
+      Tuple4(
+        StringInput,
+        OptionalInput(UpdateInput),
+        OptionalInput(ChangedInput),
+        NonEmptyList(MemberScoreInput)
+      ),
+      LongOutput
+    )
+
+  final def zAddWithIncr =
+    RedisCommand(
+      "ZADD",
+      Tuple5(
+        StringInput,
+        OptionalInput(UpdateInput),
+        OptionalInput(ChangedInput),
+        IncrementInput,
+        NonEmptyList(MemberScoreInput)
+      ),
+      DoubleOutput
```

Comment:
Just one more change: add `OptionalOutput(DoubleOutput)` to cover cases when output is `$-1\n\r` (which is equivalent to `nil` in redis-cli).
```
ZADD myzset XX INCR 3.2 "three"
$-1

```